src/main/java/core/basesyntax/strategy/handler/PurchaseHandler.java

```diff
@@ -0,0 +1,4 @@
+package core.basesyntax.strategy.handler;
+
+public class PurchaseHandler extends DecrementHandler {
```

Comment:
The PurchaseHandler class should override the handle method to ensure that it checks the result balance before saving it in the Storage. The balance after the purchase operation must be positive, and you should throw a RuntimeException if it's negative, as per the checklist item 'Handling Purchase operation.'